Feature/share post likes count(wr9 65)

HeoJeongHyeon opened the pull request:
# Pull Request Template

## 📝 PR 설명

- 좋아요 개수 및 사용자 좋아요 상태 반환 api

## 🧪 테스트

- [x] 단위 테스트 추가/수정
- [x] 테스트 커버리지 확인
- [x] 소나클라우드 품질 게이트 통과

## ✅ 체크리스트

- [x] 코드 컨벤션을 준수했나요?
- [x] 불필요한 코드나 주석이 없나요?
- [x] 브랜치 네이밍 컨벤션을 따랐나요?
- [x] 관련 문서를 업데이트했나요?

## 🤝 관련 이슈

- closes #91

HeoJeongHyeon committed:
test 좋아요 테스트

- controller
- service
- entity
- dto

HeoJeongHyeon committed:
feat 좋아요 기능 dto

- response

HeoJeongHyeon committed:
refactor 주석삭제

HeoJeongHyeon committed:
feat 좋아요 기능

controller
service
Repository

HeoJeongHyeon committed:
Merge branch 'dev' into feature/share-post-likes-count(WR9-65)

yhd1101 commented on src/main/java/io/crops/warmletter/domain/share/repository/SharePostLikeRepository.java:
```diff
@@ -10,6 +13,13 @@
+            "(SELECT s.isLiked FROM SharePostLike s " +
+            " WHERE s.sharePostId = :sharePostId AND s.memberId = :memberId)) " +
+            "FROM SharePostLike spl " +
+            "WHERE spl.sharePostId = :sharePostId")
```
서브쿼리 말고 CASE 활용하는건 어떠실까요? 그냥 의견입니다!

HeoJeongHyeon replied:
CASE 활용 방식도 괜찮은 것 같습니다.
현재 존재하지 않는 데이터가 Null 값으로 처리되어 Response 객체에 매핑되는데 boolean타입에 자동으로 false가 들어가기 때문에 이렇게 사용해도 충분하겠지 라는 생각이었습니다. 근데 이왕이면 명확하게 True Or False로 반환되게 해주는 것이 좋을 것 같아서 수정해보겠습니다~~ !

HeoJeongHyeon committed:
refactor 쿼리 수정

- 기존 서브쿼리를 통해 사용자의 데이터가 존재하지않으면 Null값으로 Response 객체에 매핑되는 과정에서 False로 자동 변환되게 해놨었음.
- 팀원분의 의견으로 CASE를 활용하면 True/False와 같은 명확한 boolean값을 제공해주므로, 이왕이면 명확하게 해결해주는 것이 더 좋은 프로그래밍 방향이라고 생각해서 리펙토링하였음.
